src/python/grpcio_tests/tests/unit/_signal_client.py

```diff
@@ -72,13 +73,46 @@ def main_streaming(server_target):
         assert False, _ASSERTION_MESSAGE
 
 
+def main_unary_with_exception(server_target):
+    """Initiate an RPC with wait_for_ready set and no server backing the RPC."""
+    channel = grpc.insecure_channel(server_target)
+    try:
+        channel.unary_unary(UNARY_UNARY)(_MESSAGE, wait_for_ready=True)
+    except KeyboardInterrupt:
+        sys.stderr.write("Running signal handler.\n")
+        sys.stderr.flush()
+
+    # This call should not hang.
+    channel.close()
+
+
+def main_streaming_with_exception(server_target):
+    """Initiate an RPC with wait_for_ready set and no server backing the RPC."""
+    channel = grpc.insecure_channel(server_target)
+    try:
+        channel.unary_stream(UNARY_STREAM)(_MESSAGE, wait_for_ready=True)
+    except KeyboardInterrupt:
+        sys.stderr.write("Running signal handler.\n")
+        sys.stderr.flush()
+
+    # This call should not hang.
+    channel.close()
+
+
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description='Signal test client.')
     parser.add_argument('server', help='Server target')
+    parser.add_argument('arity', help='Arity', choices=('unary', 'streaming'))
```

Comment:
This is a general question: why we are testing two "arity" here instead of all four of them? Do you think it is necessary?

Reply:
Yes. The point is to exercise both the "segregated" and "integrated" code paths.